homeassistant/components/fritz/common.py

```diff
@@ -0,0 +1,270 @@
+
+from homeassistant.const import (
+    CONF_DEVICES,
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_USERNAME,
+)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.util import dt as dt_util, get_local_ip
+
+from .const import (
+    ATTR_HOST,
+    DEFAULT_HOST,
+    DEFAULT_PORT,
+    DEFAULT_USERNAME,
+    DOMAIN,
+    ERROR_CONNECTION_ERROR,
+    TRACKER_SCAN_INTERVAL,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+Device = namedtuple("Device", ["mac", "ip", "name"])
+
+
+CONFIG_SCHEMA = vol.Schema(
+    vol.All(
+        cv.deprecated(DOMAIN),
+        {
+            DOMAIN: vol.Schema(
+                {
+                    vol.Required(CONF_DEVICES): vol.All(
+                        cv.ensure_list,
+                        [
+                            vol.Schema(
+                                {
+                                    vol.Optional(CONF_HOST): cv.string,
+                                    vol.Optional(CONF_PORT): cv.port,
+                                    vol.Required(CONF_USERNAME): cv.string,
+                                    vol.Required(CONF_PASSWORD): cv.string,
+                                }
+                            )
+                        ],
+                    )
+                }
+            )
+        },
+    ),
+    extra=vol.ALLOW_EXTRA,
+)
+
+SERVICE_SCHEMA = vol.Schema({vol.Required(ATTR_HOST): cv.string})
+
+
+class FritzBoxTools:
+    """FrtizBoxTools class."""
+
+    def __init__(
+        self,
+        hass,
+        password,
+        username=DEFAULT_USERNAME,
+        host=DEFAULT_HOST,
+        port=DEFAULT_PORT,
+    ):
+        """Initialize FritzboxTools class."""
+        self._cancel_scan = None
+        self._device_info = None
+        self._devices: Dict[str, Any] = {}
+        self._unique_id = None
+        self.connection = None
+        self.error = None
+        self.fritzhosts = None
+        self.fritzstatus = None
+        self.ha_ip = None
+        self.hass = hass
+        self.host = host
+        self.password = password
+        self.port = port
+        self.success = None
+        self.username = username
+
+    async def async_setup(self):
+        """Wrap up FritzboxTools class setup."""
+        return await self.hass.async_add_executor_job(self.setup)
+
+    def setup(self):
+        """Set up FritzboxTools class."""
+        try:
+            self.connection = FritzConnection(
+                address=self.host,
+                port=self.port,
+                user=self.username,
+                password=self.password,
+                timeout=60.0,
+            )
+
+            self.fritzstatus = FritzStatus(fc=self.connection)
+            self._unique_id = self.connection.call_action("DeviceInfo:1", "GetInfo")[
+                "NewSerialNumber"
+            ]
```

Comment:
Can this change?  If not we should use the store value instead of fetching it every time

Reply:
Done